### ISSUE TITLE
Add spacing abstraction layer

### ISSUE THREAD

**Issue by danifornells** [haiticss/haiticss#64]
This week I'd read an excellent article about spacings in design systems:
https://medium.com/eightshapes-llc/space-in-design-systems-188bcbae0d62

After some thoughts, I believe it would be nice to have a spacing abstraction layer with those concepts, and remove all spacing related modifiers on objects.

What do you think?
@nucliweb @area73 @arnau-pujol @judabelli @d-asensio

**Comment by judabelli** [comment]
I'm agree. In my opinion this is a more accurated way of having spaces that we actually had.  :)

**Comment by arnau-pujol** [comment]
@danifornells , I agree with you as well. In my opinion, this would be a good improvement to apply to the project. Is more consistent and solve some troubles that we found in the past.

**Comment by area73** [comment]
I like the idea.
I really like to have a visual  tool to see all the insets, stacks etc like it is shown in the article on "Teach a System of Space Visually" section.

What worries me is about "Collisions like Line Height Systematically" and adding negative margins (that really scare me if other developers have to use it, it can end up in a big mess)

**Comment by nucliweb** [comment]
Great Idea!

**Comment by d-asensio** [comment]
Hey guys! Do you think that we can consider this as done? @danifornells @arnau-pujol @judabelli Or is better to keep the issue opened for future improvements?
